### ISSUE TITLE
Intercept timeout errors?

### ISSUE THREAD

**Issue by testerez** [axios/axios#1174]
I use `interceptors.response.use` to log details about the request when for every request we do (error of success). It's very useful to have it in our logs to investigate on issues.
But when we get a timeout error, those interceptors are not invoked so all we get in our logs is:`Error: timeout of 3000ms exceeded`
Is there some kind of global event or something that I could use to add context to timeout errors?

### Context

- axios version: *v0.16.2*
- Environment: *node v8.5.0*

**Comment by nickknissen** [comment]
When a timeout happens it will return an error code with the value `ECONNABORTED`
https://github.com/axios/axios/blob/b7d8d126dad7341addcd7b063f7b261955ec9152/lib/adapters/xhr.js#L95

This is more or less how i do:
```js
axios.interceptors.response.use(
  config => config,
  (error) => {
    if (error.response.status === 408 || error.code === 'ECONNABORTED') {
      console.log(`A time happend on url ${error.config.url}`)
    }
    return Promise.reject(error);
  },
);
```

**Comment by davidlukerice** [comment]
The `ECONNABORTED` code is also used when the request is manually aborted (https://github.com/axios/axios/blob/503418718f669fcc674719fd862b355605d7b41f/lib/adapters/xhr.js#L71). Is there a good way to differentiate between the server request timeouts and manual aborts?

We're not currently using `.abort()` anywhere, so that check works for us for now, but I could see a bug crop up if we later had to add in a manual request cancel and forgot about our assumption of using that code only for the timeout case.

**Comment by wasnlosdu** [comment]
But just a bit below that is:
https://github.com/axios/axios/blob/503418718f669fcc674719fd862b355605d7b41f/lib/adapters/xhr.js#L89

Is a timeout handler - the **Error Code** is the same but it's a different handler.

**Comment by davidlukerice** [comment]
Right, that works when using the individual callbacks, but we're just catching any errors via 
```
try {
  await axios( ... );
} catch(error) {
  // determine if error was from a timeout or not
}
```
Promise type error handling doesn't seem fully supported at the moment.

**Comment by virtualLast** [comment]
So the catch statement will trigger if we get a 408 response from the server, or is it not fully supported still?
